logic/extpeers.go

```diff
@@ -150,6 +150,9 @@ func GetExtClientByPubKey(publicKey string, network string) (*models.ExtClient,
 
 // CreateExtClient - creates an extclient
 func CreateExtClient(extclient *models.ExtClient) error {
+	// lock because we need unique IPs and having it concurrent makes parallel calls result in same "unique" IPs
+	addressLock.Lock()
```

Comment:
this lock is premature.  It should be only around the  functionality that needs mutual exclusion, namely GetUniqueAddress and GetUniqueAddress6

Reply:
i believe that approach wouldn't solve the problem that needs addressing, because making the "getting" of the addresses mutually exclusive alone doesn't account for the "setting" part. Getting the unique ip and setting it should be atomic, they can't be separate from each other, otherwise we can still have race conditions. In the case we don't make it atomic, there can be two sequential (synchronized) calls to getting addresses (say, A and B), but if the second one (B) gets the address lock before the first one (A) sets the node/client in the db, then B will have the same "unique" address that A got.